sharding/nodesCoordinator/hashValidatorShuffler.go

```diff
@@ -747,14 +732,12 @@ func (rhs *randHashShuffler) UpdateShufflerConfig(epoch uint32) {
 		"maxNodesToShufflePerShard", rhs.activeNodesConfig.NodesToShufflePerShard,
 	)
 
-	rhs.flagBalanceWaitingLists.SetValue(epoch >= rhs.enableEpochsHandler.BalanceWaitingListsEnableEpoch())
-	log.Debug("balanced waiting lists", "enabled", rhs.flagBalanceWaitingLists.IsSet())
-
-	rhs.flagStakingV4Step3.SetValue(epoch >= rhs.stakingV4Step3EnableEpoch)
-	log.Debug("staking v4 step3", "enabled", rhs.flagStakingV4Step3.IsSet())
-
-	rhs.flagStakingV4Step2.SetValue(epoch >= rhs.stakingV4Step2EnableEpoch)
-	log.Debug("staking v4 step2", "enabled", rhs.flagStakingV4Step2.IsSet())
+	epochSubscriberHandler, ok := rhs.enableEpochsHandler.(core.EpochSubscriberHandler)
+	if !ok {
+		log.Error("UpdateShufflerConfig: could not cast enableEpochsHandler to EpochSubscriberHandler")
+		return
+	}
+	epochSubscriberHandler.EpochConfirmed(epoch, 0)
```

Comment:
This is **not ok**.
Under no circumstance we keep computational work with wrong flags in this component and correct flags in other components.

This code + the one you changed in nodes coordinator are time critical at epoch change and we should not do cast s every time.
Why not use an interface which extends `EnableEpochsHandler` with added `EpochConfirmed` if this what you are expecting to use?

Reply:
good suggestion, updated to use a different interface